### ISSUE TITLE
Inconsistent signature in EnableTokenAcquisitionToCallDownstreamApi

### ISSUE THREAD

**Issue by filipgoris** [Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2#446]
I found an inconsistency in the documentation/behavior of `EnableTokenAcquisitionToCallDownstreamApi` when used with ASP.Net Core 3.1 server-side Blazor.

### This issue is for a: (mark with an `x`)
```
- [ ] bug report -> please search issues before submitting
- [ ] feature request
- [x] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

### The issue was found for the following scenario:

Please add an 'x' for the scenario(s) where you found an issue

1. Web app that signs in users
   1. [ ] with a work and school account in your organization: [1-WebApp-OIDC/1-1-MyOrg](../blob/master/1-WebApp-OIDC/1-1-MyOrg)
   1. [ ] with any work and school account: [/1-WebApp-OIDC/1-2-AnyOrg](../blob/master/1-WebApp-OIDC/1-2-AnyOrg)
   1. [ ] with any work or school account or Microsoft personal account: [1-WebApp-OIDC/1-3-AnyOrgOrPersonal](../blob/master/1-WebApp-OIDC/1-3-AnyOrgOrPersonal)
   1. [ ] with users in National or sovereign clouds [1-WebApp-OIDC/1-4-Sovereign](../blob/master/1-WebApp-OIDC/1-4-Sovereign)
   1. [ ] with B2C users [1-WebApp-OIDC/1-5-B2C](../blob/master/1-WebApp-OIDC/1-5-B2C)
1. Web app that calls Microsoft Graph
   1. [x] Calling graph with the Microsoft Graph SDK: [2-WebApp-graph-user/2-1-Call-MSGraph](../blob/master/2-WebApp-graph-user/2-1-Call-MSGraph)
   1. [ ] With specific token caches: [2-WebApp-graph-user/2-2-TokenCache](../blob/master/2-WebApp-graph-user/2-2-TokenCache)
   1. [ ] Calling Microsoft Graph in national clouds: [2-WebApp-graph-user/2-4-Sovereign-Call-MSGraph](../blob/master/2-WebApp-graph-user/2-4-Sovereign-Call-MSGraph)
1. [ ] Web app calling several APIs [3-WebApp-multi-APIs](../blob/master/3-WebApp-multi-APIs)
1. [ ] Web app calling your own Web API
   1. [ ] with a work and school account in your organization: [4-WebApp-your-API/4-1-MyOrg](../blob/master/4-WebApp-your-API/4-1-MyOrg)
   1. [ ] with B2C users: [4-WebApp-your-API/4-2-B2C](../blob/master/4-WebApp-your-API/4-2-B2C)
   1. [ ] with any work and school account: [4-WebApp-your-API/4-3-AnyOrg](../blob/master/4-WebApp-your-API/4-3-AnyOrg)
1. Web app restricting users
   1. [ ] by Roles: [5-WebApp-AuthZ/5-1-Roles](../blob/master/5-WebApp-AuthZ/5-1-Roles)
   1. [ ] by Groups: [5-WebApp-AuthZ/5-2-Groups](../blob/master/5-WebApp-AuthZ/5-2-Groups)
1. [ ] Deployment to Azure
1. [ ] Other (please describe)

### Repro-ing the issue

**Repro steps**

- Create a new server-side Blazor project with ASP.Net Core 3.1.
- Install nuget package Microsoft.Identity.Web 1.4.0
- Follow steps in [2-1-Call-MSGraph](https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/tree/master/2-WebApp-graph-user/2-1-Call-MSGraph) to enable calling MS Graph.

IntelliSense does not accept the `initialScopes` parameter in `.EnableTokenAcquisitionToCallDownstreamApi(initialScopes)` in Startup.cs.

The signature is:

`EnableTokenAcquisitionToCallDownstreamApi(Action<ConfidentialClientApplicationOptions> configureConfidentialClientApplicationOptions, IEnumerable<string>? initialScopes = null)` 

Starting from a .Net 5.0 server-side Blazor project and adding Microsoft.Identity.Web 1.4.0 does produce the documented behavior, and an `EnableTokenAcquisitionToCallDownstreamApi` implementation that accepts `IEnumerable<string>` as its sole parameter.

**Expected behavior**

I would expect `EnableTokenAcquisitionToCallDownstreamApi` to behave the same in .Net Core 3.1 and .Net 5.0
-- or-- 
the difference to be documented

**Actual behavior**

The behavior is different in ASP.Net Core 3.1 and .Net 5, but not documented (afaik)

### OS and Version?
Windows 10.0.19041

### Versions
- ASP.NET Core 3.1 and 5.0
- Microsoft.Identity.Web 1.4.0
- VS Pro 2019 16.8.3
- .Net SDK 5.0.101

> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

**Comment by jmprieur** [comment]
@filipgoris 
This is not a question of .NET Core 3.1 or .NET 5, but of which override of `AddMicrosoftIdentityWebApp` you use:

- When you use the override of `AddMicrosoftIdentityWebApp` with a IConfiguration, you can use two overrides of  `EnableTokenAcquisitionToCallDownstreamApi` (the one passing-in the scopes, and the one passing in a delegate to initialize the MSAL.NET `ConfidentialClientApplicationOptions` and the scopes)
- When you use the override of `AddMicrosoftIdentityWebApp` with a delegate initializing the `MicrosoftIdentityOptions`, you can only use the override of  `EnableTokenAcquisitionToCallDownstreamApi` with a delegate to initialize the MSAL.NET `ConfidentialClientApplicationOptions` and the scopes, as the configuration is not known.

The builders are different depending on the override of AddMicrosoftIdentityWebApp, (or AddMIcrosoftIdentityWebApi) that you use


![image](https://user-images.githubusercontent.com/13203188/102192569-3d1c3f80-3ebb-11eb-901f-eabc3b17e0af.png)

**Comment by filipgoris** [comment]
@jmprieur 

I used the Startup.cs in [2-3-Multi-Tenant](https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/tree/master/2-WebApp-graph-user/2-3-Multi-Tenant) initially in the ASP.Net Core 3.1 project, which uses the override of `AddMicrosoftIdentityWebApp` with a delegate, and indeed also calls 'EnableTokenAcquisitionToCallDownstreamApi` with a delegate, not a string array.

In the ASP.Net 5.0 project I used the default server-side Blazor template which calls `AddMicrosoftIdentityWebApp` with an IConfiguration. 

Didn't realize they were different overrides with different implementations of `EnableTokenAcquisitionToCallDownstreamApi`. Thx for the explanation.
